pkg/model/iam/iam_builder.go

```diff
@@ -747,105 +781,61 @@ func AddAWSLoadbalancerControllerPermissions(p *Policy, resource stringorslice.S
 				"elasticloadbalancing:AddTags",
 				"elasticloadbalancing:RemoveTags",
 			),
-			Resource: resource,
+			Resource: stringorslice.String("*"),
 			Condition: Condition{
 				"StringEquals": map[string]string{
-					"aws:ResourceTag/elbv2.k8s.aws/cluster": clusterName,
+					"aws:ResourceTag/elbv2.k8s.aws/cluster": p.clusterName,
 				},
 			},
 		},
-		&Statement{
-			Effect: StatementEffectAllow,
-			Action: stringorslice.Of(
-				"elasticloadbalancing:DescribeTags",
-				"elasticloadbalancing:DescribeTargetGroupAttributes",
-				"elasticloadbalancing:DescribeRules",
-				"elasticloadbalancing:DescribeTargetHealth",
-				"elasticloadbalancing:DescribeListenerCertificates",
-				"elasticloadbalancing:CreateRule",
-			),
-			Resource: resource,
-		},
 	)
 }
 
-func AddClusterAutoscalerPermissions(p *Policy, clusterName string) {
-	resource := stringorslice.Slice([]string{"*"})
-	p.Statement = append(p.Statement,
-		&Statement{
-			Effect: StatementEffectAllow,
-			Action: stringorslice.Of(
-				"autoscaling:SetDesiredCapacity",
-				"autoscaling:TerminateInstanceInAutoScalingGroup",
-			),
-			Resource: resource,
-			Condition: Condition{
-				"StringEquals": map[string]string{
-					"autoscaling:ResourceTag/KubernetesCluster": clusterName,
-				},
-			},
-		},
-		&Statement{
-			Effect: StatementEffectAllow,
-			Action: stringorslice.Of(
-				"autoscaling:DescribeAutoScalingGroups",
-				"autoscaling:DescribeAutoScalingInstances",
-				"autoscaling:DescribeLaunchConfigurations",
-			),
-			Resource: resource,
-		},
+func AddClusterAutoscalerPermissions(p *Policy) {
```

Comment:
Perhaps these would be better written as receivers on `*Policy`? They could then return `p` for fluent use.

Reply:
Since all these Add* are becoming component-specific, I am considering moving these to be with the subject. Today this is in `pkg/model/components/addonmanifests/<addon>/iam.go`